blocks/editable/index.js

```diff
@@ -507,17 +507,34 @@ export default class Editable extends Component {
 	 * @param {KeydownEvent} event The keydow event as triggered by tinyMCE.
 	 */
 	onKeyDown( event ) {
+		const dom = this.editor.dom;
+		const rootNode = this.editor.getBody();
+
 		if (
-			this.props.onMerge && (
-				( event.keyCode === BACKSPACE && this.isStartOfEditor() ) ||
-				( event.keyCode === DELETE && this.isEndOfEditor() )
-			)
+			( event.keyCode === BACKSPACE && this.isStartOfEditor() ) ||
+			( event.keyCode === DELETE && this.isEndOfEditor() )
 		) {
-			const forward = event.keyCode === DELETE;
+			if ( ! this.props.onMerge && ! this.props.onRemove ) {
+				return;
+			}
+
 			this.fireChange();
-			this.props.onMerge( forward );
+
+			const forward = event.keyCode === DELETE;
+
+			if ( this.props.onMerge ) {
+				this.props.onMerge( forward );
+			}
+
+			const isEmpty = dom.isEmpty( rootNode );
+			if ( isEmpty && this.props.onRemove ) {
```

Comment:
Minor: If `dom.isEmpty` is non-trivial ([sorta appears so](https://github.com/tinymce/tinymce/blob/master/src/core/src/main/js/dom/Empty.js)), we could achieve a minor optimization here with conditional short-circuiting by reordering the condition to test the truthy `onRemove` first:

```js
if ( this.props.onRemove && dom.isEmpty( rootNode ) ) {
```